src/git.ts

```diff
@@ -1,6 +1,8 @@
 import {exec} from '@actions/exec';
 import {PathLike} from 'fs';
 
+import {execWithStdoutCap} from './process-utils';
+
 export class Repo {
```

Comment:
If this class is inherited or its objects are used somewhere, maybe you could consider writing an `interface` for it? You can then `implement` that interface for this class.

Reply:
What would be the benefit of writing an interface over just exposing the class itself? It is incredibly unlikely to ever have more than one implementation.

I'm not sure how interfaces work in TypeScript, is there some added benefit apart from hiding the actual implementing class?

Reply:
If multiple instances of `Repo` is created, you can use the interface to define their type. That is the only benefit I can see. :sweat_smile:

Reply:
Perhaps you are thinking of pure JavaScript? In TypeScript, a class should itself be a type, yes? I.e. this should be valid TypeScript:

```typescript
const repo: git.Repo = new git.Repo("some/path");
const otherRepo: git.Repo = new git.Repo("other/path");
```

Or am I completely missing something here? I'm just going by the assumption that TypeScript follows "normal" type system conventions, I haven't actually read anything about this.

Reply:
> In TypeScript, a class should itself be a type, yes?

Right! TypeScript follows the same conventions.

TypeScript interfaces are usually written when:

1. Use type checking at compile time. 
2. And like you said, to maintain consistency across multiple implementations.

> I'm not sure how interfaces work in TypeScript, is there some added benefit

The interfaces are not transpiled to JavaScript so you only use it for type checking at compile time. It just confirms that you are passing correct type objects/parameters to the methods. Apart from this, I don't think so there are any other benefits.

Reply:
I'm with you on 2., but 1. still confuses me. TypeScript does type checking at compile time regardless of interfaces, that's it's whole thing :). If I for example do this:

```typescript
const repo: git.Repo = new git.Repo("/some/path");
repo.nonExistingMethod();
```

I get this error on compilation:

```
src/main.ts:25:8 - error TS2339: Property 'nonExistingMethod' does not exist on type 'Repo'.

25   repo.nonExistingMethod();
          ~~~~~~~
```

I think what you're thinking of is if I have some arbitrary object (for example when parsing JSON), which I'd like to impose types on. Then one can use an interface to declare what that object will contain after being parsed from JSON. I have a bunch of that in [sorald.ts](https://github.com/SpoonLabs/sorald-buildbreaker/blob/main/src/sorald.ts). But there's afaik no need to define an interface for a type that you've declared yourself, as all of the typing information is already available to the TypeScript compiler.

Reply:
> I get this error on compilation:

src/main.ts:25:8 - error TS2339: Property 'nonExistingMethod' does not exist on type 'Repo'.

25   repo.nonExistingMethod();
          ~~~~~~~

Oh, okay. Then interfaces don't make sense for your case. I was not sure if TS would throw an error.

> I think what you're thinking of is if I have some arbitrary object (for example when parsing JSON), which I'd like to impose types on. 

Yes, I meant that too. Interfaces are quite useful there.